common/templates/context.go

```diff
@@ -957,6 +957,16 @@ func (s Slice) Append(item interface{}) (interface{}, error) {
 	}
 }
 
+func (s Slice) Del(index int) (interface{}, error) {
+	if index >= len(s) {
+		return nil, errors.New("Index out of bounds.")
+	}
+
+	result := reflect.ValueOf(&s).Elem()
+	result = reflect.AppendSlice(result.Slice(0, index), result.Slice(index+1, result.Len()))
+	return result.Interface(), nil
```

Comment:
Reflection is not needed (and `Append` and `AppendSlice` use it gratuitously.)
```suggestion
	// Clip the capacity of the left slice to force a copy on append, for consistency with Append and AppendSlice
	// (which return copies.)
	return append(s[:index:index], s[index+1:]...)
```
should be sufficient.